components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Response.kt

```diff
@@ -44,10 +44,28 @@ data class Response(
      * A [Body] returned along with the [Request].
      *
      * **The response body can be consumed only once.**.
+     *
+     * @param stream the input stream from which the response body can be read.
+     * @param contentType optional content-type as provided in the response
+     * header. If specified, an attempt will be made to look up the charset
+     * which will be used for decoding the body. If not specified, or if the
+     * charset can't be found, UTF-8 will be used for decoding.
      */
     open class Body(
-        private val stream: InputStream
+        private val stream: InputStream,
+        contentType: String? = null
     ) : Closeable, AutoCloseable {
+
+        @Suppress("TooGenericExceptionCaught")
+        private val charset = contentType?.let {
+            val charset = it.substringAfter("charset=")
+            try {
+                Charset.forName(charset)
+            } catch (e: Exception) {
+                Charsets.UTF_8
+            }
+        } ?: Charsets.UTF_8
```

Comment:
Interestingly this document says that the default for HTTP 1.1 is `ISO-8859-1`. I wonder what okhttp does?

https://www.w3.org/International/articles/http-charset/index#charset

Reply:
@pocmo okhttp uses the charset specified in the content-type header and if absent tries byte order mark (https://en.wikipedia.org/wiki/Byte_order_mark), then falls back to UTF-8.

Reply:
@pocmo I think we should stick with UTF-8 as fallback. Here's a good discussion why this is better (for XHR and JSON): https://github.com/AsyncHttpClient/async-http-client/issues/1439

Reply:
TL;DR: There's a bit of a misunderstanding/contradiction in the spec and the default charset depends on the mimetype. For application/JSON this should be UTF-8.

Reply:
Alright :+1: